great_expectations/expectations/registry.py

```diff
@@ -355,10 +355,13 @@ def get_expectation_impl(expectation_name: str):
         )
         expectation_name = renamed[expectation_name]
 
-    if expectation_name not in _registered_expectations:
+    expectation: Type[Expectation] | None = _registered_expectations.get(
+        expectation_name
+    )
+    if not expectation:
         raise ge_exceptions.ExpectationNotFoundError(f"{expectation_name} not found")
 
-    return _registered_expectations.get(expectation_name)
+    return expectation
```

Comment:
Here I am inverting this check to make the type-checker happy but the result is the same.